### PR TITLE
Improved JS code architecture in the extension

### PR DESCRIPTION
**Pull request by valgreens**
Improved JS code architecture in the extension and also added a random color generator for each project. Removed jQuery dependency.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,11 +26,10 @@ to:
 TODO
 ----
 
-- Add MOAAAAAAAAAAARRR nice colors!
 - Improve the CSS applied a little bit.
 - Allow configuration for the "seen" projects background color.
 - Allow configuration for the list of colors.
-- Calculate the color instead set a random one?
+- Improve color load for projects when clicking on 'more' button
```
Comment (agonzalezro) [inline README.md]: Minor comment: `.` at the end.

File: colorify.js [hunk 2]
```diff
@@ -1,53 +1,63 @@
...
+      }
+    });
+
+    document.onreadystatechange = function () {
+      if (document.readyState == "complete") {
+        self.paint();
+        document.querySelector('.js-events-pagination').onclick = function () {
+          setTimeout(function () {
+            self.paint();
+          }, 1000);
```
Comment (agonzalezro) [inline colorify.js:30]: Is this going to refresh every second? Why you didn't like the `click` event on the "Show more" button?

File: colorify.js [hunk 3]
```diff
@@ -1,53 +1,63 @@
...
+  };
+
+  colorify.paint = function () {
+    var self = this,
+        previousColor;
+
+    [].forEach.call(document.querySelectorAll('.alert'), function (element) {
+      var project = self.getProjectName(element),
+          color = self.getColorForProject(project, previousColor);
+      console.log(color, project);
```
Comment (agonzalezro) [inline colorify.js]: remove

File: colorify.js [hunk 4]
```diff
@@ -1,53 +1,63 @@
...
-  $(document).ready(function() {
-    $(".js-events-pagination").click(function() {
-      setTimeout(function() { colorify(); assignEvents(); }, 1000)
+  };
+
+  colorify.paint = function () {
+    var self = this,
+        previousColor;
+
+    [].forEach.call(document.querySelectorAll('.alert'), function (element) {
```
Comment (agonzalezro) [inline colorify.js]: Is this right? I suspect that it's some weird JS trick :D

File: colorify.js [hunk 5]
```diff
@@ -1,53 +1,63 @@
...
+
+  colorify.getProjectName = function (element) {
+    var link = [].pop.call(element.getElementsByClassName('title')[0].getElementsByTagName('a'));
+
+    return link.innerHTML.split('#')[0].split('@')[0];
+  };
+
+  colorify.getColorForProject = function (project, previousColor) {
+    if (!(project in this.colors)) {
+      var color = ('00000' + (Math.random() * (1 << 24) | 0).toString(16)).slice(-6);
```
Comment (agonzalezro) [inline colorify.js]: I would move this to a function with a meaningfully name

File: colorify.js [hunk 6]
```diff
@@ -1,53 +1,63 @@
...
     }
-    colorify();
-  });
-  assignEvents();
-}
+    return this.colors[project];
+  };
+
+  colorify.init();
+
```
Comment (agonzalezro) [inline colorify.js]: Extra line?

File: manifest.json [hunk 7]
```diff
@@ -15,7 +15,7 @@
   "content_scripts": [
     {
       "matches": ["https://github.com/*"],
-      "js": ["jquery.min.js", "colorify.js"]
+      "js": ["colorify.js"]
```
Comment (agonzalezro) [inline manifest.json:18]: Awesome! :) Thanks!

File: colorify.js (hunk 2 above)
Comment (agonzalezro) [inline colorify.js:30]: My bad. I was reading this as a `setInterval` and it didn't make a lot of sense.

**Comment by valgreens** [comment]
I've made some changes: removed `console.log` statement, improved function names for readability and rewrote `popup.js` to work without jQuery. I think it's ready to merge when you have some time to check.

**Comment by agonzalezro** [comment]
Nice one! I really like the changes, and the style is much better now as we discussed.

I continue not being sure that the distinction between projects is clear, but let's start with something!

**Pull request merged**